### DISCUSSION TITLE
2p2c miscible fluids - How to remove everything which comes with the water?

### DISCUSSION THREAD

**Discussion by 1runer** [idaholab/moose#25734] [Q&A Modules: Porous Flow]
Hello,

I would like to simulate water injection and production of water+CH4 in a two-phase two-component miscible system
and I hope @cpgr or @WilkAndy can help me with a question about DiracKernels.

The idea is to inject pure water into an injection well and to produce with a certain flow rate water + CH4 (in all phases).

I am not quite sure how to set the DiracKernels to remove a certain (known) rate of water (e.g. 5 kg/s) 
and to remove the entire (unknown) CH4 component (as a gas and dissolved in water) so that the total mass fraction of CH4 is basically zero?

Is the way to track the amount per dt (SumQuantityUO = borehole_total_outflow_gas) correct? 

Please see the attached code snipped:

```
[DiracKernels]
 [inject_H2O]
    type = PorousFlowPolyLineSink
    variable = pgas
    SumQuantityUO = borehole_total_inflow_water
    # mass_fraction_component = 0
    point_file = borehole_vert.bh
    fluid_phase = 0
    p_or_t_vals = '-1e9 1e9'
    fluxes = '-1 -1' # kg/m.s * 25 m length * 0.2 m borehole diameter --> 5 kg/s
  []
  [produce_H2O]
    type = PorousFlowPolyLineSink
    variable = pgas
    SumQuantityUO = borehole_total_outflow_water
    point_file = borehole_vert2.bh
    p_or_t_vals = '-1e9 1e9' # kg/m.s * 25 m length * 0.2 m borehole diameter --> 5 kg/s
    fluxes = '1 1'
  []
  [produce_CH4]
    type = PorousFlowPolyLineSink
    SumQuantityUO = borehole_total_outflow_gas
    fluxes = ??????
    p_or_t_vals = 0.0
    mass_fraction_component = 1
    point_file = borehole_vert2.bh
    variable = zi
  []
[]
```

**Comment by cpgr** [comment]
Ooops, I missed this one. Did you figure it out?

**Reply by 1runer** [reply]
I simply multiplied the flow rates by the relative permeability of the phase to get a rate. However, I am not sure if this is the correct/best solution. I thought about something like an outflowBC instead of a Dirac, but this would require an explicitly meshed cross sectional area which we don't have.
@cpgr do you have a better idea in mind how to remove fluid with a certain rate  
and at the same time remove all CH4 dissolved in the water and which is present as gas phase? 
In reality the CH4 gas would just migrate towards the surface in the wellbore?
And for sure, how to track the mass rates?

**Reply by cpgr** [reply]
That's probably what I would have tried too

**Reply by 1runer** [reply]
Is there something comparable like an outflowBC which allows for degassing within a wellbore?
